tests/main/interfaces-location-control/task.yaml

```diff
@@ -29,6 +29,11 @@ prepare: |
 
 restore: |
     systemctl stop test-snapd-location-control-provider.service
+    # provider.py was started from a snap by the test-snapd-location-control-provider.service unit,
+    # normally it should be enough to kill the whole unit, however since that process
+    # was started from a snap, it got moved to a separate snap-specific transient
+    # scope, thus it is no longer part of the unit's cgroup
+    pkill -f provider.py || true
```

Comment:
Works in the short term. Ideally it looks like the 'provider' that we run here, could actually be a service, i.e. declared as a `daemon` in snap.yaml?

Reply:
Yeah, there is an XXX comment in this test already. I don't have time to play with this atm but can revisit it later, fwtw that should unblock the invariant check PR.